lib/companies/scheduler.ex

```diff
@@ -0,0 +1,26 @@
+defmodule Companies.Scheduler do
+  use GenServer
+
+  def start_link({_mod, _func, _argument_list, _interval} = state) do
+    GenServer.start_link(__MODULE__, state)
+  end
+
+  @impl true
+  def init({_mod, _func, _argument_list, interval} = state) do
+    schedule_work(interval)
+
+    {:ok, state}
+  end
+
+  @impl true
+  def handle_info(:work, {mod, func, argument_list, interval} = state) do
+    apply(mod, func, argument_list)
+    schedule_work(interval)
+
+    {:noreply, state}
+  end
+
+  defp schedule_work(interval) do
+    Process.send_after(self(), :work, interval)
```

Comment:
In this implementation, each deployment (with restart) will shift the task execution, thus if it is often deployed, a task can be performed very rarely. 
Although I think this is not critical for us, it will be possible to add task execution immediately after the start of the gen_server.

Reply:
Having the task run on start would be bad if app is crashing, for example. OTP will continue restarting it and we'll create load on other people's servers.

Having this delayed is intentional to make sure we make as least amount of requests as possible.

Also, it would be an interesting case if we had a problem in this checking jobs code. Then this OTP process will fail, restart, and scheduled once again. In case of immediate run - we'll be accessing few first jobs immediately, and again and again.

Reply:
We could potentially leverage a log to prevent excess execution based on last entry timestamp.

Reply:
I meant not executing checker code directly in gen_server init, and once sending an asynchronous message to yourself, same as by timeout.

But if the execution frequency is not important, then let's just skip it.